lib/gps.js

```diff
@@ -0,0 +1,394 @@
+//var IS_TEST_MODE = !!process.env.IS_TEST_MODE;
+var Board = require("./board");
+var events = require("events");
+var util = require("util");
+var Pin = require("./pin");
+
+var priv = new Map();
+
+var Breakouts = {
+
+  /*
+   * https://www.adafruit.com/products/746
+   */
+  adafruitUltimateGPS: {
+    receiver: {
+      value: "FGPMMOPA6H"
+    }
+  }
+
+};
+
+// GPS Antenna Modules
+var Receivers = {
+
+  /*
+   * http://www.gtop-tech.com/en/product/LadyBird-1-PA6H/MT3339_GPS_Module_04.html
+   */
+  FGPMMOPA6H: {
+    // Later, when we add logging that code will go here
+    chip: {
+      value: "MT3339"
+    }
+  }
+
+};
+
+// GPS chips
+var Chips = {
+
+  /*
+   * http://www.mediatek.com/en/products/connectivity/gps/mt3339/
+   */
+  MT3339: {
+    baud: {
+      value: 9600
+    },
+    configure: {
+      value: function(callback) {
+        callback();
+      }
+    },
+    frequency: {
+      value: function(freq) {
+        var state = priv.get(this);
+
+        // Enforce maximum frequency of 10hz
+        if (freq < 100) {
+          freq = 100;
+        }
+
+        this.sendCommand("$PMTK220,"+String(freq));
+        state.frequency = freq;
+      }
+    },
+    restart: {
+      // Reboot the receiver
+      value: function(coldRestart) {
+
+        if (coldRestart === true) {
+          this.sendCommand("$PMTK103");
+        } else {
+          this.sendCommand("$PMTK101");
+          setTimeout(function() {
+            this.sendCommand("");
+          }.bind(this), 1000);
+        }
+
+      }
+    }
+  }
+
+};
+
+
+/**
+ *
+ * @constructor
+ *
+ * @param {Object} opts Options: pin(s), chip, receiver, breakout, precision, serialport, freq
+ *
+ * Sample initialization
+ *
+ *    new five.GPS({ pins: {rx: 10, tx: 11});
+ *
+ */
+
+function GPS(opts) {
+
+  var breakout, receiver, chip, state;
+
+  if (!(this instanceof GPS)) {
+    return new GPS(opts);
+  }
+
+  // Allow users to pass in a 2 element array for tx and rx pins
+  if (Array.isArray(opts)) {
+    opts = { pins: {tx: opts[0], rx: opts[1], onOff: opts[2] }};
+  }
+
+  if (typeof opts.pins === "undefined") {
+    opts.pins = {};
+  }
+
+  Board.Component.call(
+    this, this.opts = Board.Options(opts)
+  );
+
+  breakout = opts.breakout || "adafruitUltimateGPS";
+
+  // Get user overrides for chip and receivers
+  receiver = opts.receiver;
+  chip = opts.chip;
+
+  if (!receiver && Breakouts[breakout].receiver) {
+    receiver = Breakouts[breakout].receiver.value;
+  }
+
+  if (!chip && Breakouts[breakout].chip) {
+    chip = Breakouts[breakout].chip.value;
+  }
+
+  // If a receiver was defined or derived but chip was not
+  if (receiver && !chip && Receivers[receiver].chip) {
+    chip = Receivers[receiver].chip.value;
+  }
+
+  // Allow users to pass in custom chip types
+  chip = typeof chip === "string" ?
+    Chips[chip] : this.opts.chip;
```

Comment:
Please don't do this: `this.opts`. `chip` should be a property directly on `this` (and same for all others)